KarhooSDK/Service/Payment/AdyenPaymentMethods/KarhooAdyenPaymentMethodsInteractor.swift

```diff
@@ -46,8 +46,8 @@ final class KarhooAdyenPaymentMethodsInteractor: AdyenPaymentMethodsInteractor {
 
     private func handle(_ response: Result<HttpResponse>, interactorCallback: CallbackClosure<DecodableData>) {
         switch response {
-        case .failure(let error): interactorCallback(.failure(error: error))
-        case .success(let response): interactorCallback(.success(result: DecodableData(data: response.data)))
+        case .failure(let error): interactorCallback(.failure(error: error.error))
+        case .success(let result): interactorCallback(.success(result: DecodableData(data: result.result.data)))
```

Comment:
don't really like this `result.result` bit. also the warnings need to be sorted out. what if we go with this?
```
case .failure(let error, _): interactorCallback(.failure(error: error))
case .success(let result, _): interactorCallback(.success(result: DecodableData(data: result.data)))
```

Reply:
then we will wet warning that accesing tuplets in that way is deprecated :/

Reply:
We could rename the companion value identifier of success case